x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -113,9 +116,12 @@ class RollupShardIndexer {
                 null,
                 Collections.emptyMap()
             );
-            this.timestampField = searchExecutionContext.getFieldType(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+            this.timestampField = (DateFieldMapper.DateFieldType) searchExecutionContext.getFieldType(config.getTimestampField());
             this.timestampFormat = timestampField.docValueFormat(null, null);
             this.rounding = config.createRounding();
+            this.millisecondsSupplier = DateFieldMapper.Resolution.MILLISECONDS.equals(this.timestampField.resolution())
```

Comment:
Wouldn't `timestampField.resolution().roundDownToMillis()` work here?

Reply:
Yes I didn't see it. Thanks.